Run3Detector/analysis/wip/dataCuts.py

```diff
@@ -0,0 +1,275 @@
+            return
+        if self.debug:
+            print(channels)
+        return event
+
+
+    # Removes events that have more than a single hit per layer, ensuring particle travels in straight line
+    def singleHitPerLayer(self, event):
+        layers = event.layer
+        # Sets have no repeated values so if the lenghts are not the same, there is some repeated value
+        if len(set(layers)) != len(layers):
+            return
+        if self.debug:
+            print(layers)
+        return event
+
+
+    #MuonSelection is base on NeHa's idea
+    #this will not work in run 591 because there is not much hit in the end pannels
+    def muonSelection(self,event, height_threshold):
+        
+        height_list = event.height
+        chan_list = event.chan
+
+        # if there is no hit on the end pannels, then we can't claims that this 
+        # event is muon event
+        if 71 not in chan_list:
+            return
+        elif 75 not in chan_list:
+            return
+        frontPannelHeight = []
+        backPannelHeight = []
+        #She claim if both end pannals has a big hit(>1.1V), then it is muon event
+        for chan, height in zip(chan_list,height_list):
+
+            if chan == 71:  
+                frontPannelHeight.append(height)
+            elif chan == 75:
+                backPannelHeight.append(height)
+            
+            if any(x >= height_threshold for x in frontPannelHeight) and any(x >= height_threshold for x in backPannelHeight):
+                return event
+            else:
+                return
+
+    def npeCheck(self,event):
+        nPE_list = event.nPE
+        DAQNum = event.DAQEventNumber
+        if min(nPE_list) >= 60:
+            #print(DAQNum)
+            return event
+        else:
+            return None
+
+
+    #this method is used with ThreeInLine()
+    def layerCheck(self,lists):
+        i = 0 
+        for list in lists:
+            if len(set(list)) >= 3: #return true if it contain 3 unique layers(3 in a line)
+                i += 1
+        return i
+
+
+    
+    def ThreeInLine(self,event):
+        row_list = event.row
+        column_list = event.column
+        layer_list = event.layer
+
+        # four in a line is very rare impossible to find in run 591
+        # use three in a line instead!
+        if len(set(layer_list)) >= 3:  #three in a line
+            pass
+        else:
+            return
+
+        
+        # the list at below save the information about the pulse passing through difference layers
+        # For example, List0 is used for saving the infomation pulse passing through the position of
+        # row is three and column is one. "0" correspond to the position of the 0th channal, whose location
+        # is at the third row and first column.
+
+        #numbers at below are index number. So 3rd(index) row mean the 4th row in my notation
+        list0 = [] # correspond to 3rd row, 0st column
```

Comment:
Can this be made into a 2d array to avoid defining so many arrays? Could also help condense the large elif block below.